specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+}
+```
+
+## .NET/ WinRT
+```c#
+private bool _isClientCertificateRequested = false;
+void ClientCertificateRequestedCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+  try
+  {
+    if (!_isClientCertificateRequested)
+    {
+      webView.CoreWebView2.ClientCertificateRequested += delegate (
+      object sender, CoreWebView2ClientCertificateRequestedEventArgs args)
+      {
+        // Developer can obtain a deferral for the event so that the WebView2
+        // doesn't examine the properties we set on the event args until
+        // after the deferral completes asynchronously.
+        CoreWebView2Deferral deferral = args.GetDeferral();
+
+        System.Threading.SynchronizationContext.Current.Post((_) =>
+        {
+          using (deferral)
+          {
+            List<CoreWebView2ClientCertificate> certificatesList = args.MutuallyTrustedCertificates;
+            if (certificatesList.Count() > 0)
+            {
+              var dialog = new ClientCertificateSelectionDialog(
+                           title: "Client Certificate Selection",
+                           host: args.Host,
+                           port: args.Port,
+                           client_cert_list: certificatesList);
+              if (dialog.ShowDialog() == true)
+              {
+                args.SelectedCertificate = (CoreWebView2ClientCertificate)dialog.CertificateDataBinding.SelectedItem;
+                args.ClientCertificateRequestResponseState = CoreWebView2ClientCertificateRequestResponseState.ContinueWithCertificate;
+              }
+              else
+              {
+                args.ClientCertificateRequestResponseState = CoreWebView2ClientCertificateRequestResponseState.ContinueWithoutCertificate;
+              }
+            }
+            else
+            {
+              args.ClientCertificateRequestResponseState = CoreWebView2ClientCertificateRequestResponseState.ContinueWithoutCertificate;
+            }
+          }
+        }, null);
+      };
+      _isClientCertificateRequested = true;
+    }
+  }
+  catch (NotImplementedException exception)
+  {
+    MessageBox.Show(this, "Client Certificate Requested Failed: " + exception.Message, "Client Certificate Requested");
+  }
+}
+```
+# Remarks
+
+# API Notes
+
+# API Details
+See [API Details](#api-details) section below for API reference.
+
+## Win32 C++
+```cpp
+interface ICoreWebView2_3;
+interface ICoreWebView2ClientCertificate;
+interface ICoreWebView2CertificateAuthorityList;
+interface ICoreWebView2ClientCertificateIssuerChainList;
+interface ICoreWebView2ClientCertificateList;
+interface ICoreWebView2ClientCertificateRequestedEventArgs;
+interface ICoreWebView2ClientCertificateRequestedEventHandler;
+
+[v1_enum]
+/// Question to reviewers if COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE
+/// enum should be replaced with Cancel and Handled properties like below to match
+/// with other WebView2 API conventions ????
+/// * Cancel = true -> AbortRequest
```

Comment:
Yes, I was going to leave the same question; can we use the existing Handled and Cancel patterns?